lib/eslint.js

```diff
@@ -123,35 +134,37 @@ function run(console, runOptions) {
   function inclusionBasedFileListBuilder(includePaths) {
     // Uses glob to expand the files and directories in includePaths, filtering
     // down to match the list of desired extensions.
-    return function(extensions) {
+    return async function(extensions) {
       var analysisFiles = []
 
-      includePaths.forEach(function(fileOrDirectory, i) {
+      await Promise.all(includePaths.map(async function(fileOrDirectory, i) {
         if (/\/$/.test(fileOrDirectory)) {
           // if it ends in a slash, expand and push
           var filesInThisDirectory = glob.sync(fileOrDirectory + "/**/**")
-          prunePathsWithinSymlinks(filesInThisDirectory).forEach(function(file, j) {
-            if (!isFileIgnoredByLibrary(file) && isFileWithMatchingExtension(file, extensions)) {
+          prunePathsWithinSymlinks(filesInThisDirectory).forEach(async function(file, j) {
+            const isIgnored = await isFileIgnoredByLibrary(file)
+            if (!isIgnored && isFileWithMatchingExtension(file, extensions)) {
               analysisFiles.push(file)
             }
           })
         } else {
-          if (
-            !isFileIgnoredByLibrary(fileOrDirectory) &&
+          const isIgnored = await isFileIgnoredByLibrary(fileOrDirectory)
+          if(
+            !isIgnored &&
             isFileWithMatchingExtension(fileOrDirectory, extensions)
           ) {
             analysisFiles.push(fileOrDirectory)
           }
         }
-      })
+      }))
 
       return analysisFiles
     }
   }
 
   function overrideOptions(userConfig) {
     if (userConfig.configPath) {
-      options.configFile = codeDir + "/" + userConfig.configPath
+      options.overrideConfigFile = codeDir + "/" + userConfig.configPath
```

Comment:
is this intentional ?

Reply:
Yes, this is the new name of the `configFile` attribute in the [options object](https://github.com/eslint/eslint/blob/main/lib/eslint/eslint.js#L191).